Firestore/Source/Remote/FSTRemoteStore.mm

```diff
@@ -627,7 +630,7 @@ - (void)handleHandshakeError:(NSError *)error {
 
 - (void)handleWriteError:(NSError *)error {
   // Only handle permanent error. If it's transient, just let the retry logic kick in.
-  if (![FSTDatastore isPermanentWriteError:error]) {
+  if (![FSTDatastore isPermanentWriteError:error previousError:[_writeStream lastError]]) {
```

Comment:
I was going to suggest that we should have similar logic for the watch stream to not retry, but thinking harder I think:

1) It's not particularly harmful to the backend to retry indefinitely (we'll end up using the max backoff reconnect rate).
2) The more important thing is to notify the user that there's something wrong rather than silently retrying without any feedback.  For the watch stream, this should be addressed by the improved "offline" logging I added (https://github.com/firebase/firebase-js-sdk/blob/5e82b606c8e9f7e4e27ca321529b4fb36791dab6/packages/firestore/src/remote/online_state_tracker.ts#L178) which I still need to port to iOS / Android.  And for the write stream, we can probably solve this via the improved logging described in b/80200799.

All of this is to say that I think for simplicity I'd recommend that we let it retry indefinitely and remove the previousError tracking / conditional logic.

Reply:
Done, thanks.